File: Station_Frontend/lib/models/Station.dart
import 'price.dart';
import 'service.dart';
class Station {
  final int id;
  final String nom;
  final String logoUrl;
  final String adresse;
  final String ville;
  final double latitude;
  final double longitude;
  List<Price> prices;
  List<Service> services;

  Station({
    required this.id,
    required this.nom,
    required this.logoUrl,
    required this.adresse,
    required this.ville,
    required this.latitude,
    required this.longitude,
    this.prices = const [],
    this.services = const [],
  });

  // Factory method to convert JSON data to Station object
  factory Station.fromJson(Map<String, dynamic> json) {
    return Station(
      id: json['id'] as int,
      nom: json['nom'] as String,
      logoUrl: json['logoUrl'] as String,
      adresse: json['adresse'] as String,
      ville: json['ville'] as String,
      latitude: json['latitude'] as double,
      longitude: json['longitude'] as double,

    );
  }

}

File: GestionStation/src/main/java/com/ensa/gestionstation/model/Station.java
package com.ensa.gestionstation.model;

import lombok.*;

import javax.persistence.*;
@Entity
@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Station {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String nom;
    private String logoUrl;
    private String adresse;
    private String ville;
    private float latitude;
    private float longitude;


}
